packages/core/src/shared/telemetry/spans.ts

```diff
@@ -219,8 +219,8 @@ export class TelemetrySpan<T extends MetricBase = MetricBase> implements Span<T>
                 result: getTelemetryResult(err),
                 reason: getTelemetryReason(err),
                 reasonDesc: getTelemetryReasonDesc(err),
-                requestId: getRequestId(err),
-                httpStatusCode: getHttpStatusCode(err),
+                ...(!this.state.requestId ? { requestId: getRequestId(err) } : {}),
+                ...(!this.state.httpStatusCode ? { httpStatusCode: getHttpStatusCode(err) } : {}),
```

Comment:
Making this change to allow `requestId` and `httpStatusCode` to be set explicitly before the `stop` method is invoked. Otherwise it will silently get removed if `err` is not the correct format.